### ISSUE TITLE
Serialization uses the class provided, but can result in a CCE if it is not

### ISSUE THREAD

**Issue by minborg** [OpenHFT/Chronicle-Wire#277]
The following test fails because no exception is thrown. Instead, an instance is created using fields from another class (sic!).

```
package net.openhft.chronicle.wire.issue;

import net.openhft.chronicle.core.pool.ClassAliasPool;
import net.openhft.chronicle.wire.SelfDescribingMarshallable;
import net.openhft.chronicle.wire.WireType;
import org.junit.Ignore;
import org.junit.Test;

public class Issue277 {

    @Ignore("https://github.com/OpenHFT/Chronicle-Wire/issues/277")
    @Test(expected = IllegalArgumentException.class)
    public void reproduce() {
        // This is needed in order to reproduce the problem
        ClassAliasPool.CLASS_ALIASES.addAlias(Data1.class);

        // This should throw a RuntimeException as Data1.class != Data2.class
        Data2 o2 = WireType.TEXT.fromString(Data2.class, "!Data1 {\n" +
                "  name: Tom,\n" +
                "  age: 25,\n" +
                "  address: \"21 high street, Liverpool\"\n" +
                "}\n");

        System.out.println("o2 = " + o2);
    }

    private static class Data1 extends SelfDescribingMarshallable {
        String name;
        int age;
        String address;
    }

    private static class Data2 extends SelfDescribingMarshallable {
        String name;
        int age;
        String address;
    }

}
```

**Comment by peter-lawrey** [comment]
I have added some Javadoc to clarify this behaviour

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.21ea43](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.21ea43), [BOM-2.21ea123](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.21ea123)
